lib/modules/manager/composer/artifacts.ts

```diff
@@ -18,26 +18,51 @@ import {
 import { getRepoStatus } from '../../../util/git';
 import * as hostRules from '../../../util/host-rules';
 import { regEx } from '../../../util/regex';
+import { GitTagsDatasource } from '../../datasource/git-tags';
 import { PackagistDatasource } from '../../datasource/packagist';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
 import type { AuthJson, ComposerLock } from './types';
 import {
   extractConstraints,
+  findGithubToken,
   getComposerArguments,
   getPhpConstraint,
+  isGithubFineGrainedPersonalAccessToken,
+  isGithubPersonalAccessToken,
+  isGithubServerToServerToken,
   requireComposerDependencyInstallation,
+  takePersonalAccessTokenIfPossible,
 } from './utils';
 
 function getAuthJson(): string | null {
   const authJson: AuthJson = {};
 
-  const githubCredentials = hostRules.find({
+  const githubToken = findGithubToken({
     hostType: 'github',
     url: 'https://api.github.com/',
   });
-  if (githubCredentials?.token) {
+
+  const gitTagsGithubToken = findGithubToken({
+    hostType: GitTagsDatasource.id,
+    url: 'https://github.com',
+  });
+
+  const selectedGithubToken = takePersonalAccessTokenIfPossible(
+    githubToken,
+    gitTagsGithubToken
+  );
+  if (selectedGithubToken) {
+    if (isGithubPersonalAccessToken(selectedGithubToken)) {
+      logger.debug('Using GitHub Personal Access Token');
+    } else if (isGithubServerToServerToken(selectedGithubToken)) {
+      logger.debug('Using GitHub Server-to-Server token');
+    } else if (isGithubFineGrainedPersonalAccessToken(selectedGithubToken)) {
+      logger.debug('Using GitHub Fine-grained Personal Access Token');
+    } else {
+      logger.debug('Using unknown GitHub token type');
+    }
```

Comment:
i think those log messages should be moved to the  take personal access token function.

Reply:
Done, it allowed a more precise logging and simplified the unit tests to get code coverage on those lines.

Thanks.